extensions/amp-story/1.0/amp-story.js

```diff
@@ -1550,6 +1554,22 @@ export class AmpStory extends AMP.BaseElement {
     this.setOrientationAttribute_(isLandscape, isLandscapeSupported);
   }
 
+  /**
+   * Handles resize events and sets CSS variables.
+   * @param {!Object} size including new width and height
+   * @private
+   */
+  onResize_(size) {
+    const {height, width} = size;
+    if (height === 0 && width === 0) {
+      return;
+    }
+    this.storeService_.dispatch(Action.SET_PAGE_SIZE, {height, width});
+    setImportantStyles(this.win.document.documentElement, {
```

Comment:
Please check if `dvh` is supported before performing this.

Reply:
Good idea, wdyt if we don't even add the observer if dvh is supported?

Reply:
That's even better